CMakeLists.txt

```diff
@@ -128,6 +131,64 @@ if( NOT WITH_REFERENCE_BACKEND AND
 	message( FATAL_ERROR "At least one backend should be enabled")
 endif()
 
+# Adding backends for tests only
+set( AVAILABLE_TEST_BACKENDS "" )
+if( WITH_REFERENCE_BACKEND )
+	list( APPEND AVAILABLE_TEST_BACKENDS "reference" )
+endif()
+if( WITH_OMP_BACKEND )
+	list( APPEND AVAILABLE_TEST_BACKENDS "reference_omp" )
+endif()
+if( WITH_HYPERDAGS_BACKEND )
+	list( APPEND AVAILABLE_TEST_BACKENDS "hyperdags" )
+endif()
+if( WITH_NONBLOCKING_BACKEND )
+	list( APPEND AVAILABLE_TEST_BACKENDS "nonblocking" )
+endif()
+if( WITH_BSP1D_BACKEND )
+	list( APPEND AVAILABLE_TEST_BACKENDS "bsp1d" )
+endif()
+if( WITH_HYBRID_BACKEND )
+	list( APPEND AVAILABLE_TEST_BACKENDS "hybrid" )
+endif()
+
+# Enabling reference(+omp) backend if bsp1d is active
+if( WITH_BSP1D_BACKEND )
+	set( WITH_REFERENCE_BACKEND ON )
+	set( WITH_OMP_BACKEND ON )
+	message( STATUS "Enabling compilation of reference backend: required by bsp1d" )
+endif()
+
+# Enabling reference(+omp) backend if hybrid is active
+if( WITH_HYBRID_BACKEND )
+	set( WITH_OMP_BACKEND ON )
+	set( WITH_OMP_BACKEND ON )
+	message( STATUS "Enabling compilation of reference_omp backend: required by hybrid" )
+endif()
+
+# Enabling reference(+omp) backend if non-blocking is active
+if( WITH_NONBLOCKING_BACKEND )
+	set( WITH_REFERENCE_BACKEND ON )
+	set( WITH_OMP_BACKEND ON )
+	message( STATUS "Enabling compilation of reference backend: required by nonblocking" )
+endif()
+
+# Enabling backend required by hyperdags (if not)
+if( WITH_HYPERDAGS_BACKEND )
+	if( WITH_HYPERDAGS_USING STREQUAL "reference" )
+		set( WITH_REFERENCE_BACKEND ON )
+		message( STATUS "Enabling reference backend for hyperdags" )
+	elseif(WITH_HYPERDAGS_USING STREQUAL "reference_omp" )
+		set( WITH_OMP_BACKEND ON )
+		message( STATUS "Enabling reference_omp backend for hyperdags" )
+	elseif(WITH_HYPERDAGS_USING STREQUAL "nonblocking" )
+		set( WITH_NONBLOCKING_BACKEND ON )
+		message( STATUS "Enabling non-blocking backend for hyperdags" )
+	else()
+		message( FATAL_ERROR "Unrecognised backend for hyperdags: ${WITH_HYPERDAGS_USING}.\nAvailable options: reference, reference_omp, nonblocking" )
+	endif()
+endif()
```

Comment:
Should this code block not appear before line 136? (Or is it the idea that the tests of the backends hyperdags depend on are not enabled unless explicitly asked for?)

Reply:
It is precisely for that idea. 
We allow the user to test the backends he asked for only, and then enable the compilation for the necessary ones.